### PR TITLE
fixed animateToItem return type as Future.wait returns a Future<List<T>>

### PR DESCRIPTION
**Pull request by thinkmobilede**
when executing animateToItem an error is thrown that the return type of the function does not match the return type of Future.wait()

### CODE REVIEW

**Comment by thinkmobilede** [comment]
I signed it!

File: packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart [hunk 1]
```diff
@@ -79,7 +79,7 @@ class FixedExtentScrollController extends ScrollController {
   /// The returned [Future] resolves when the animation completes.
   ///
   /// The `duration` and `curve` arguments must not be null.
-  Future<Null> animateToItem(int itemIndex, {
+  Future<List<Null>> animateToItem(int itemIndex, {
```
Comment (thinkmobilede) [inline packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart]: The function returns the result of `Future.wait(futures);` which is declared as `Future<List<T>>`, so the return type has to match. Before the execution was throwing an error when finished.

**Comment by Hixie** [comment]
I think the signature we had before (returning `Future<Null>`, or maybe even `Future<void>`) is probably clearer to the caller. We can convert the `Future.wait` future to that easily enough by doing:
```dart
  return Future.wait<Null>(...).then<void>((List<Null> values) => null);
```

**Comment by Hixie** [comment]
Looks like this was actually fixed in https://github.com/flutter/flutter/pull/20618, so I'm going to close this PR. Thanks for your contribution!

**Pull request closed without merging**